test/functional-portable/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,389 @@
+/*
+Copyright 2025 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade_test
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/radius-project/radius/test/rp"
+	"github.com/radius-project/radius/test/testcontext"
+	"github.com/radius-project/radius/test/testutil"
+	"github.com/stretchr/testify/require"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	radiusNamespace   = "radius-system"
+	preUpgradeJobName = "pre-upgrade"
+	testTimeout       = 5 * time.Minute
+
+	relativeChartPath = "../../../deploy/Chart"
+)
+
+func Test_PreflightContainer_FreshInstall(t *testing.T) {
+	ctx := testcontext.New(t)
+
+	image, tag := getPreUpgradeImage()
+
+	// Clean up any existing installation using helm
+	t.Log("Cleaning up any existing Radius installation")
+	cleanupCommand := []string{"helm", "uninstall", "radius", "--namespace", radiusNamespace, "--ignore-not-found"}
+	_ = exec.Command(cleanupCommand[0], cleanupCommand[1:]...).Run() // Ignore errors during cleanup
+
+	// Use local registry image for testing the pre-upgrade functionality
+	t.Log("Installing Radius with preflight enabled using helm")
+	installCommand := []string{
+		"helm", "install", "radius", relativeChartPath,
```

Comment:
for now, these are helm install commands. we can (and probably should) change this to `rad upgrade` and `rad rollback` in the future @ytimocin